code/game/objects/items/storage/boxes/job_boxes.dm

```diff
@@ -53,19 +60,45 @@
 	new /obj/item/radio/off(src)
 
 /obj/item/storage/box/survival/proc/wardrobe_removal()
-	if(!isplasmaman(loc) && !isvox(loc)) //We need to specially fill the box with plasmaman gear, since it's intended for one	//SKYRAT EDIT: && !isvox(loc)
+	//SKYRAT EDIT BEGIN: setup for nitrogen-breathing quirk
+	var/nitrogen_lungs = CheckNitrogenLungs()
+	//SKYRAT EDIT END
+	if(!isplasmaman(loc) && !(isvox(loc) || isvoxprimalis(loc) || nitrogen_lungs)) //We need to specially fill the box with plasmaman gear, since it's intended for one	//SKYRAT EDIT: check for Vox/nitrogen breathers
 		return
 	var/obj/item/mask = locate(mask_type) in src
 	var/obj/item/internals = locate(internal_type) in src
 	//SKYRAT EDIT ADDITION START - VOX INTERNALS - Vox mimic the above and below behavior, removing the redundant mask/internals; they dont mimic the plasma breathing though
-	if(!isvox(loc))
+	if(!(isvox(loc) || isvoxprimalis(loc) || nitrogen_lungs))
 		new /obj/item/tank/internals/plasmaman/belt(src)
 	else
-		new /obj/item/tank/internals/nitrogen/belt/emergency(src)
+		if(internal_type == /obj/item/tank/internals/emergency_oxygen/engi) //engineers get extended tanks
+			new /obj/item/tank/internals/nitrogen/belt/(src)
+		else
+			new /obj/item/tank/internals/nitrogen/belt/emergency(src)
 	//SKYRAT EDIT ADDITION END - VOX INTERNALS
 	qdel(mask) // Get rid of the items that shouldn't be
 	qdel(internals)
 
+/obj/item/storage/box/survival/proc/CheckNitrogenLungs() //SKYRAT EDIT: A utility function to look for those with the Nitrogen Breather quirk or other sources of nitrogen lungs.
+	if(ishuman(loc))
+		var/mob/living/carbon/human/checked_human = loc
+		var/qrk
+		if(checked_human.client)
+			if(checked_human.client.prefs)
+				for(qrk in checked_human.client.prefs.all_quirks)
+					if(qrk == "Nitrogen Breather") //hardcoded- this isn't great but whatever
+						return TRUE
```

Comment:
This will be much cleaner.
```suggestion
		if(checked_human?.client?.prefs)
			for(var/quirk in checked_human?.client?.prefs?.all_quirks)
				if(quirk == "Nitrogen Breather") //hardcoded- this isn't great but whatever
					return TRUE
```